src/pystow/utils.py

```diff
@@ -27,12 +28,85 @@
 logger = logging.getLogger(__name__)
 
 
+class HexDigestError(ValueError):
+    """Thrown if the hashsums do not match expected hashsums."""
+
+    def __init__(self, offending_hexdigests: Collection[Tuple[str, str]]):
+        self.offending_hexdigests = offending_hexdigests
+
+    def __str__(self):
+        return "\n".join((
+            "Hexdigest of downloaded file does not match the expected ones!",
+            *(
+                f"\tactual: {actual} vs. expected: {expected}"
+                for actual, expected in self.offending_hexdigests
+            )
+        ))
+
+
+def get_offending_hexdigests(
+    destination: Path,
+    chunk_size: int = 64 * 2 ** 10,
+    verbose: bool = True,
+    hexdigests: Optional[Mapping[str, str]] = None,
+) -> Collection[Tuple[str, str]]:
+    """
+    Check a file for hash sums.
+
+    :param destination:
+        The file path.
+    :param chunk_size:
+        The chunk size for reading the file.
+    :param hexdigests:
+        The expected hexdigests as (algorithm_name, expected_hex_digest) pairs.
+    :param verbose:
```

Comment:
why verbose? can't this just be handled by the user choosing the logging level?

Reply:
:+1:

Reply:
Good then please remove the verbose argument